### DISCUSSION TITLE
Return different status for successful/failed patch/delete

### DISCUSSION THREAD

**Discussion by JUVOJustin** [directus/directus#12782] [Feature Requests]
- When sending `PATCH` Requests for an unknown object no error nor a status code other than 200 is returned
- When sending `DELETE` Requests no error nor a status code other than 200 is returned

I am using Directus 9 RC 63 in Docker deployment. 

I am currently building an [Integromat](https://docs.integromat.com/apps/structure-blocks/api/handling-responses#valid) integration and would like to verify the directus api responses. I was able to verify `PATCH` requests by checking if a body is returned. That does not seem to work for the `DELETE`  endpoint since it never returnes anything.

I dont know if this is intentional, but maybe a status code change would be an easy way of seperate the non-successful from successful requests.

**Comment by maxbritto** [comment]
I confirm this issue on Directus 9.0.1.
I have been trying to do a PATCH on an item and it kept returning 200 with no error, even in the server logs but nothing was changed.
I was accidentaly wrapping the object content in a `data` json object (the same one directus is using when sending us content). That is a quick fix, but I've spent quite some time trying to understand what was wrong with my directus instance 😁
My guess is that the PATCH method only uses the fields it recognize and just ignores the unknown fields. It just thought "data" was a field I was trying to update on my item and ignored it because there is no such field in this item type. Maybe just logging in the server logs the ignored fields would be helpful

**Comment by rijkvanzanten** [comment]
This is currently by design to prevent users from scraping what items exist in a given collection. That's the same reason why GET requests will return 403s for both not-found and no-permissions 👍🏻

**Reply by maxbritto** [reply]
That makes sense, we do not want responses to give out too much information about our internal data model. 
What about using the logs without changing the returned response ?

**Reply by rijkvanzanten** [reply]
That could work! We'll have to think through how to configure that, as I'd imagine you might not want that additional noise busying out the output at all times 🤔 (also seeing that any scraper bot will cause a lot of those additional logs to be added)

**Reply by maxbritto** [reply]
If scraper bots do use POST/PATCH requests, maybe we could only log those errors when using the `debug` `LOG_LEVEL `. 
this would allow developers to quickly detect their mistakes in their dev environment, without logging on production servers. 
Would that be an acceptable solution ?

**Reply by rijkvanzanten** [reply]
Yeah! That works 👍🏻

**Reply by rijkvanzanten** [reply]
I'd also like to look into ways to configure pino to only log certain "groups" rather than just by level. I remember the `debug` package having a similar piece of functionality through it's `DEBUG=scope:*` env vars 🤔 (I'd like to be able to only trace the DB queries, but not the above mentioned warnings for example)

**Comment by jaads** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
